src/components/About/WarningBanner.js

```diff
@@ -0,0 +1,78 @@
+import _ from 'lodash';
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  Headline,
+  List,
+  MessageBanner,
+} from '@folio/stripes-components';
+
+import { isVersionCompatible } from '../../discoverServices';
+import css from './About.css';
+
+const WarningBanner = ({
+  interfaces,
+  modules,
+}) => {
+  const allInterfaces = useMemo(() => {
+    const modulesArray = _.flatten(_.values(modules));
+
+    return modulesArray.reduce((prev, curr) => Object.assign(prev, curr.okapiInterfaces), {});
+  }, [modules]);
```

Comment:
@uladislausamets thank you for trying out `useMemo`. Unfortunately this still won't work as expected because `useMemo` is not doing deep compare under the hood. If you put `console.log` right inside this `useMemo` you will notice that it's still being executed on every rerender. So this is actually worst that what we had before.

One way to solve it would be to use something similar to:

https://github.com/sandiiarov/use-deep-compare#usedeepcomparememo

another would be to do: `JSON.stringify(modules)` which sometimes beats the purpose of caching (because it's actually slower).

I think would introduce this new dependency to stripes core:

https://github.com/sandiiarov/use-deep-compare

and try to use `import { useDeepCompareMemo } from "use-deep-compare";`

I'm sure it will be helpful to have it around in other places.

Reply:
`useDeepCompareMemo` helped to decrease the count of functions executions

Reply:
useDeepCompareMemo looks like a hack to me, is there a way when props aren't updated on every render?

Reply:
@aliaksei-chumakou another approach here could be to use `React.memo` on the entire component. I think deep comparing (maybe with https://lodash.com/docs/4.17.15#isEqual) would have to be used there too.

Reply:
but it would clean up this code a bit because `useMemo` would not have to be used. Maybe that's the way to do it.

Reply:
@uladislausamets sorry for raising these questions originally but I think it was a good exercise to see different approaches with these kind of issues.

Reply:
I also tried `React.memo` and it isn't so effective as `useDeepCompareMemo`